jbmc/src/java_bytecode/java_object_factory.cpp

```diff
@@ -794,77 +794,69 @@ void java_object_factoryt::gen_nondet_struct_init(
   // * Not if the object has already been initialised by our caller, in which
   //   case they will set `skip_classid`
   // * Not if we're re-initializing an existing object (i.e. update_in_place)
-
-  bool skip_special_string_fields = false;
-
-  if(!is_sub &&
-     !skip_classid &&
-     update_in_place != update_in_placet::MUST_UPDATE_IN_PLACE)
+  // * Always if it has a string type (special case).
```

Comment:
:pick: I don't think `(special case)` is necessary here. Rather an explanation of *why* strings are a special case would be useful.

Reply:
I added a few more lines, with a reference to the function that initialises string fields. Any further documentation should probably go to that function.